### PR TITLE
Add Hong Kong calendar.  Factor out the Christmas/Boxing Day Sunday s…

### PR DESCRIPTION
**Pull request by nedlowe**
…hift as its used in the UK and in HK.  Add a test to Poland as an example to ensure it *doesn't* shift countries which don't shift Christmas.  Add complicated Sunday shift for the third day of CNY - I would rather loop this than hardcode it, but I didn't want to change the core approach until it had been discussed.

refs #

For information, read and make sure you're okay with the [CONTRIBUTING document](https://github.com/novafloss/workalendar/blob/master/CONTRIBUTING.rst#adding-new-calendars).

- [x] Tests with a significant number of years to be tested for your calendar.
- [x] Docstrings for the Calendar class and specific methods.
- [x] Calendar country / label added to the README.rst file,
- [x] Changelog amended with a mention like: "Added ``<country>`` by ``@pseudo`` (#)"

### CODE REVIEW

**Comment by brunobord** [comment]
It's weird to have all these changes at once, but it looks fine: thanks to our tests, it's obvious that you didn't break a thing!

Don't worry about the fact that it's "hardcoded" in some way, we can refactor this bit later on.

File: workalendar/core.py [hunk 1]
```diff
@@ -327,6 +327,19 @@ def get_whit_sunday(self, year):
     def get_corpus_christi(self, year):
         return self.get_easter_sunday(year) + timedelta(days=60)
 
+    def shift_christmas_boxing_days(self, year):
```
Comment (brunobord) [inline workalendar/core.py]: I think you can add a docstring here, for more clarity. it's not a blocking point, though

File: workalendar/core.py [hunk 2]
```diff
@@ -327,6 +327,19 @@ def get_whit_sunday(self, year):
...
+        christmas = date(year, 12, 25)
+        boxing_day = date(year, 12, 26)
+        results = []
+        if christmas.weekday() in self.get_weekend_days():
+            shift = self.find_following_working_day(christmas)
+            results.append((shift, "Christmas Shift"))
+            results.append((shift + timedelta(days=1), "Boxing Day Shift"))
```
Comment (brunobord) [inline workalendar/core.py]: I'd like to see the following label here: ``"{} Shift".format(self.boxing_day_label)``
In some countries, Boxing Day is named "St Stephen's" or "The Day After Christmas", for example

File: workalendar/core.py [hunk 3]
```diff
@@ -327,6 +327,19 @@ def get_whit_sunday(self, year):
...
+        elif boxing_day.weekday() in self.get_weekend_days():
+            shift = self.find_following_working_day(boxing_day)
+            results.append((shift, "Boxing Day Shift"))
```
Comment (brunobord) [inline workalendar/core.py]: same as above

**Comment by nedlowe** [comment]
Yeah sorry I'll try to break things up better going forward.

**Comment by nedlowe** [comment]
Hey - sorry for the delay, real life got in the way ;-)

OK I've done the requested changes, plus also fixed a couple of other edge cases I found while researching the Hong Kong calendar (see commit message for details).

**Comment by brunobord** [comment]
Thanks a lot for this contribution!

**Comment by nedlowe** [comment]
My pleasure!  I look forward to being able to contribute more.

**Pull request merged**